### DISCUSSION TITLE
how to add a new entity display like the player when you open the inventory

### DISCUSSION THREAD

**Discussion by DarkiDoki** [FabricMC/fabric-api#2644] [Mod Dev Support] [answered]
I am trying to make a mod where you can add a Steve model or Alex model and then switch between them like a morph. How do I add a model display for the selected model? The only thing I have been able to find is the PlayerEntity in the PlayerInventory... but I can't find where in the code (1.19.2) that actually makes the Entity be displayed in the position it's at in the inventory or the code that creates the displayed model in general.
I need to import a JSON model file of an Alex and Steve model and then I need the code that adds them to my GUI display. Any ideas on what I can do?

**Comment by warjort** [comment, marked as answer]
The code you are looking for is in `InventoryScreen.drawEntity()`
Not sure how useful that is for what you are doing? 
It requires a player and will use their skin along with drawing other stuff like the worn armor.

You probably want to look at `PlayerEntityRenderer` and its parent class(es).
The models are created programmatically and the vanilla game has 2 instances of this class
`default` and `slim` which are steve and alex
see the `EntityRenderers.PLAYER_RENDERER_FACTORIES`
